benchmark/benchmark_device_adjacent_difference.cpp

```diff
@@ -180,12 +190,15 @@ void run_benchmark(benchmark::State& state, const std::size_t size, const hipStr
 
 using namespace std::string_literals;
 
-#define CREATE_BENCHMARK(T, left, copy)                                    \
-    benchmark::RegisterBenchmark(("Subtract" + (left ? "Left"s : "Right"s) \
-                                  + (copy ? "Copy"s : ""s) + "<" #T ">")   \
-                                     .c_str(),                             \
-                                 &run_benchmark<T, left, copy>,            \
-                                 size,                                     \
+#define CREATE_BENCHMARK(T, left, copy)                                             \
+    benchmark::RegisterBenchmark(std::string("device_adjacent_difference"           \
+                                             "<data_type:" #T ">."                  \
+                                             "sub_algorithm_name:Subtract"          \
+                                             + std::string(left ? "Left" : "Right") \
+                                             + std::string(copy ? "Copy" : ""))     \
```

Comment:
Should this also become snake case? I'm fine with how it is currently, but maybe being more consistent is nicer.

```suggestion
                                             "sub_algorithm_name:subtract_"         \
                                             + std::string(left ? "left" : "right") \
                                             + std::string(copy ? "_copy" : ""))    \
```